tests/unit/modules/archive/archive.py

```
@@ -0,0 +1,945 @@
+
+        def get_path(*args):
+            """Get path with decompressdir prefix"""
+            return os.path.join(self.basedir.name, *args)
+
+        # File types
+        zip_path = os.path.abspath(
+            os.path.join(self.test_data_dir, "test_archive__symlinks.zip")
+        )
+        # TODO: Handle permission correctly for all users
+        # The umask is not yet handled by utils.archive, hardcode it for now
+        os.umask(2)
```

Ruff (0.16.0)

[warning] Non-octal mode

(RUF064)